### PR TITLE
LPD-26772 Add playwright test for DSM field list order

### PR DESCRIPTION
**Pull request by kresimir-coko**
Jira ticket: [LPD-26772](https://liferay.atlassian.net/browse/LPD-26772)

---

This PR extends `visualizationMode.spec.ts` to cover the following cases:

- Once fields in Table mode are added, check the right order (newest are added last)
- A field can be moved to a different position in the list using the keyboard

This way we can delete a Poshi test ([DataSetViewFields#CanChangeFieldOrder](https://github.com/liferay/liferay-portal/blob/4379629e01004cda66621cb544015ec501159672/portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFields.testcase#L942))

### CODE REVIEW

**Comment by kresimir-coko** [comment]
ci:test:frontend

File: modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts [hunk 1]
```diff
@@ -435,6 +435,87 @@ test.describe('Visualization Modes in Data Set Manager', () => {
...
+			await visualizationModesPage.selectField({
+				dataId: `${SAMPLE_OBJECT_FIELD}.${SAMPLE_OBJECT_CHILD_FIELD}`,
+				fieldName: SAMPLE_OBJECT_CHILD_FIELD,
+			});
+
+			await saveFromModal({
+				page,
+			});
+		});
+
```
Comment (dsanz) [inline modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts:479]: It'd be nice to assert the order of all fields here, and reuse this assertion in the step named "Assert that the field has moved one place up". You can use 2 approaches at least:
 - function taking the array of expected row values and assert compute the locator with 'table.orderable-table > tbody tr:nth-child(i+1)` where `i` tells the position in the array of expected values.
 - similarly but using the table locator as you do [here](https://github.com/liferay-frontend/liferay-portal/pull/4234/commits/83d210bfe6ff282c40aba3cff17fd1d98b1dbd41#diff-4fdd0b230c3151f1fc9543222b2f4adc280e7a7042036293832f82a32d5a0f86R505-R515)

**Comment by dsanz** [comment]
@kresimir-coko onde test pass, pls add a commit deleting the poshi :fire:

File: modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts (hunk 1 above)
Comment (markocikos) [reply modules/test/playwright/tests/frontend-data-set-views-web/visualizationModes.spec.ts:479]: Could we simplify even further and merge two tests into one? There is a lot of duplication in the setup, and the `Check if field is added to the last position` assertion is not destructive to UI. After it is successful, it looks like we can keep going with focus and order steps.

**Review by markocikos — COMMENTED** [review]
@kresimir-coko Tests look good! Just a comment on simplification inline.

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:frontend

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		dateCreated: '2024-06-05T10:49:03Z',
+		id: '3',
+	},
+	{
+		dateCreated: '2024-06-05T10:49:02Z',
+		id: 2,
+	},
+	{
+		dateCreated: '2024-06-05T10:49:01Z',
+		id: '1',
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts:28]: @dsanz Not a blocker for this PR, but isn't storing IDs in order CSVs wrong? I thought we started moving them to ERCs while back :thinking:

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  - keys contain itemsOrder values, will be used to call sortItems()
+  - values contain the expected order of id property after calling sortItems()
+  */
+type orderResult = {
+	[Property: string]: string;
+};
+
+const totalOrderResult: orderResult = {
+	'1,2,3,4': JSON.stringify([1, 2, 3, 4]),
+	'4,1,3,2': JSON.stringify([4, 1, 3, 2]),
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts:42]: To simplify a bit, can you remove `JSON.stringify`s in setup structures and just have in test something like
```js
.toBe(JSON.stringify(expected[itemsOrder]))
```

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		expect(
+			JSON.stringify(
+				sortItems(items, itemsOrder, useCreationDate).map((item) =>
+					Number(item.id)
+				)
+			)
+		).toBe(expected[itemsOrder])
+	);
+
+describe('sortItems()', () => {
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts:68]: That's a bit lazy description :sweat_smile:

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+/* This type represents expected order result pairs
+  - keys contain itemsOrder values, will be used to call sortItems()
+  - values contain the expected order of id property after calling sortItems()
+  */
+type orderResult = {
+	[Property: string]: string;
+};
+
+const totalOrderResult: orderResult = {
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts:40]: Could we call these structures 
- `completeSortCases`
- `partialSortCases`
- `creationDatePartialSortCases`

Following the `case` keyword, `testLoop()` can be `testCases()`.

**Review by markocikos — COMMENTED** [review]
@kresimir-coko PW tests is good to go, bunch of nits for the new unit test.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts (hunk 4 above)
Comment (kresimir-coko) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/test/utils/sortItems.spec.ts:68]: I'll keep it like that as it follows the precedent set in frontend-js-web tests. It seems like the `it` steps are the thing that has the actual description.

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:frontend

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:report:13966356

**Comment by kresimir-coko** [comment]
Testray reports no unique failures

**Comment by kresimir-coko** [comment]
ci:test:sf

**Review by markocikos — APPROVED** [review]
@kresimir-coko LGTM! :sparkles:

**Comment by kresimir-coko** [comment]
@bryceosterhaus the SF error is still happening 🤔

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:test:sf

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**